FMWK-683 Add support for restoring from archived files (#240)
aerospike/backup-go: cmd/asrestore/readme.md

So --directory path will only contain folder name.
--s3-endpoint-override is used in case you want to use minio, instead of AWS.
Any AWS parameter can be retrieved from Secret Agent.
      --s3-bucket-name string          Existing S3 bucket name
      --s3-region string               The S3 region that the bucket(s) exist in.
      --s3-profile string              The S3 profile to use for credentials.
      --s3-access-key-id string        S3 access key id. If not set, profile auth info will be used.
      --s3-secret-access-key string    S3 secret access key. If not set, profile auth info will be used.
      --s3-endpoint-override string    An alternate url endpoint to send S3 API calls to.

>       --s3-tier string                 If is set, tool will try to restore archived files with selected tier.

Reviewer: ```suggestion
      --s3-tier string                 If is set, tool will try to restore archived files to the specified tier.
```

Author: done

>       --s3-restore-poll-duration int   Time in milliseconds, how often files status will be checked on restore from archive. (default 60000)

Reviewer: Improve the flag description to something like:
> An interval for checking object status when restoring an archived object in milliseconds.

Author: added

For authentication you can use --azure-account-name and --azure-account-key, or 
--azure-tenant-id, --azure-client-id and azure-client-secret.
Any Azure parameter can be retrieved from Secret Agent.
      --azure-account-name string           Azure account name for account name, key authorization.
      --azure-account-key string            Azure account key for account name, key authorization.
      --azure-tenant-id string              Azure tenant ID for Azure Active Directory authorization.
      --azure-client-id string              Azure client ID for Azure Active Directory authorization.
      --azure-client-secret string          Azure client secret for Azure Active Directory authorization.
      --azure-endpoint string               Azure endpoint.
      --azure-container-name string         Azure container Name.

>       --azure-access-tier string            If is set, tool will try to rehydrate archived files with selected tier.

Reviewer: ```suggestion
      --azure-access-tier string            If is set, tool will try to rehydrate archived files to the specified tier.
```

Author: done
